src/index.js

```diff
@@ -686,6 +707,78 @@ export default function ({ types: t, template }) {
     return node.declaration && ( node.declaration.type === 'TypeAlias' || node.declaration.type === 'InterfaceDeclaration' )
   }
 
+  function getWrappedVariableDeclaratorInitWithTypeCheckAST(declarator) {
+    return getAssertCallExpression({
+      id: declarator.init,
+      name: t.stringLiteral(declarator.id.name || generate(declarator.id, { concise: true }).code),
+      annotation: declarator.id.typeAnnotation.typeAnnotation
+    })
+  }
+
+  function getWrappedAssignmentWithTypeCheckAST(node, typeAnnotation) {
+    return getAssertCallExpression({
+      id: node.right,
+      name: t.stringLiteral(node.left.name || generate(node.left, { concise: true }).code),
+      annotation: typeAnnotation
+    })
+  }
+
+  function findTypeAnnotationInObjectPattern(name, objectPattern, objectTypeAnnotation) {
+    if (!objectTypeAnnotation || !t.isObjectTypeAnnotation(objectTypeAnnotation)) {
+      return
+    }
+
+    for (let property of objectPattern.properties) {
+      const typeAnnotation = objectTypeAnnotation.properties.find(propType => propType.key.name === property.key.name)
+      if (!typeAnnotation) {
+        continue
+      }
+
+      if (t.isIdentifier(property.value) && name === property.value.name) {
+        return typeAnnotation.value
+      } else if (t.isObjectPattern(property.value)) {
+        const result = findTypeAnnotationInObjectPattern(name, property.value, typeAnnotation.value)
+        if (result) {
```

Comment:
is there a particular reason you don't just `return findTypeAnnotationInObjectPattern(name, property.value, typeAnnotation.value)` here?